aspnetcore/fundamentals/minimal-apis.md

```diff
@@ -1555,19 +1555,44 @@ Routes can be [CORS](xref:security/cors?view=aspnetcore-6.0) enabled using [CORS
 
 For more information, see <xref:security/cors?view=aspnetcore-6.0>
 
+<a name="openapi7"></a>
+
 ## OpenAPI
 
 An app can describe the [OpenAPI specification](https://swagger.io/specification/) for route handlers using [Swashbuckle](https://www.nuget.org/packages/Swashbuckle.AspNetCore/).
 
-The following code is a typical ASP.NET Core app with OpenAPI support:
+The following code is generated by the ASP.NET Core minimal web API template and uses OpenAPI:
+
+[!code-csharp[](minimal-apis/7.0-samples/WebMinOpenApi/Program.cs?name=snippet&highlight=1,5-6,12-13,35-36)]
+
+The OpenAPI specification defines a standard for documenting RESTful APIs so that end users and other services can examine the capabilities of an API service.
+
+ASP.NET Core provides the [`Microsoft.AspNetCore.OpenApi`](https://www.nuget.org/packages/Microsoft.AspNetCore.OpenApi/) package to interact with OpenAPI specifications for endpoints. The package acts as a link between the OpenAPI models that are defined in the `Microsoft.AspNetCore.OpenApi` package and the endpoints that are defined in Minimal APIs. The package provides an API that examines an endpoint's parameters, responses, and metadata to construct an OpenAPI annotation type that is used to describe an endpoint.
+
+`Microsoft.AspNetCore.OpenApi` is added as a PackageReference to a project file:
+
+[!code-xml[](minimal-apis/7.0-samples/WebMinOpenApi/projectFile.xml?highlight=10)]
```

Comment:
Let's add a warning callout here that the `Microsoft.AspNetCore.OpenApi` package depends on `Swashbuckle.AspNetCore` versions 6.3.1 and above.

Reply:
This statement isn't clear. `Microsoft.AspNetCore.OpenApi` doesn't directly depend on `Swashbuckle.AspNetCore`. Did you mean if they are used together? If a developer is using another OpenAPI generator, such as NSwag, it may not apply and _could_ be confusing if not worded correctly.